cpp/src/join/join.cu

```diff
@@ -13,32 +13,53 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include <cudf/table/table.hpp>
-
 #include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/dictionary/detail/update_keys.hpp>
 #include <cudf/join.hpp>
-#include <cudf/table/table_view.hpp>
+#include <cudf/table/table.hpp>
 
-#include "cudf/types.hpp"
-#include "hash_join.cuh"
+#include <join/hash_join.cuh>
+#include <join/join_common_utils.hpp>
 
 namespace cudf {
+namespace detail {
+
+cudf::table_view update_table_from_matched(cudf::table_view const& input,
+                                           cudf::table_view const& updated_table,
+                                           std::vector<size_type> const& indices)
```

Comment:
How about renaming to `scatter_columns`? I think it would be a more descriptive and generic name for this operation. Then it could go into the `table_view` header instead of being specific to join.